test/integration/tests/import_tpm.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+TPM_CC_DUPLICATE=0x14B
+TPM_CC_IMPORT=0x156
+
+source helpers.sh
+
+cleanup() {
+rm -f primary.ctx \
+          new_parent.prv new_parent.pub new_parent.ctx \
+          ipolicy.dat dpolicy.dat session.dat \
+          key.prv key.pub key.ctx \
+          dup.prv dup.pub dup.seed \
+          key2.prv key2.pub key2.ctx \
+          sym_key_in.bin
+
+    if [ "$1" != "no-shut-down" ]; then
+          shut_down
+    fi
+}
+trap cleanup EXIT
+
+start_up
+
+create_policy() {
+    tpm2_startauthsession -Q -S session.dat
+    tpm2_policycommandcode -Q -S session.dat -o $1 $2
+    tpm2_flushcontext -Q -S session.dat
+    rm session.dat
+}
+
+start_session() {
+    tpm2_startauthsession -Q --policy-session -S session.dat
+    tpm2_policycommandcode -Q -S session.dat -o $1 $2
+}
+
+end_session() {
+    tpm2_flushcontext -Q -S session.dat
+    rm session.dat
+}
+
+tpm2_createprimary -Q -a o -g sha256 -G rsa -o primary.ctx
+
+# Create new parent
+create_policy ipolicy.dat $TPM_CC_IMPORT
+tpm2_create -Q -C primary.ctx -g sha256 -G rsa -r new_parent.prv -u new_parent.pub -L ipolicy.dat -b "restricted|sensitivedataorigin|decrypt"
+# Load new parent key, only the public part
+tpm2_loadexternal -a o -u new_parent.pub -o new_parent.ctx
+
+# Create the key we want to duplicate
+create_policy dpolicy.dat $TPM_CC_DUPLICATE
+tpm2_create -Q -C primary.ctx -g sha256 -G rsa -r key.prv -u key.pub -L dpolicy.dat -b "sensitivedataorigin|decrypt"
+# Load the key
+tpm2_load -Q -C primary.ctx -r key.prv -u key.pub -o key.ctx
+
+# Extract the public part for import later
+tpm2_readpublic -c key.ctx -o dup.pub
+
+# Duplicate the key
+start_session dpolicy.dat $TPM_CC_DUPLICATE
+tpm2_duplicate -Q -C new_parent.ctx -c key.ctx -g null -p "session:session.dat" -r dup.dup -s dup.seed
+end_session
+
+rm new_parent.ctx
+
+# Load new parent key, public & private parts
+tpm2_load -Q -C primary.ctx -r new_parent.prv -u new_parent.pub -o new_parent.ctx
+
+# Import the duplicated key
+start_session ipolicy.dat $TPM_CC_IMPORT
+tpm2_import -V -C new_parent.ctx -g null -u dup.pub -d dup.dup -r dup.prv -s dup.seed -P "session:session.dat" -L dpolicy.dat
```

Comment:
Two things here:
1. I don't want to override -g to be the name algorithm or the crypto algorithm.
2. make sure this works with AES128

Reply:
1.a Sure, how about --symmetric_alg and no short form? 
1.b If the algo is null then you don't need -i (was -d)
1.C I believe null & AES128CFB (same as duplicate)

Reply:
> 1.a Sure, how about --symmetric_alg and no short form?

i'm fine with that.

> 1.b If the algo is null then you don't need -k

Can we infer the crypto algorithm from -k option or via the dup.dup file coming in?

> 1.C I believe null & AES128CFB (same as duplicate)
> 
> Drop -d (seed) and re-use -i instead, ok.

I thought -s was the seed, no?

Just be very deliberate about command line options, cleanliness and figuring out things on behalf of the user where possible. I want to avoid having a million options for tools, especially for the common case. Guess and check is a valid approach as well.

Reply:
1.a  -G / --symmetric_alg

1.b I guess if -k is specified we could assume aes else null. That would get rid of the need for --symmetric_alg. I can do it this way if you wish

1.c Yes, -s is seed (my bad). The duplicate was specified by -d but is now specified -i